modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/AbstractMetadataArgumentResolver.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleMessage;
+import org.mule.api.transformer.DataType;
+import org.mule.extension.api.runtime.ContentType;
+import org.mule.extension.api.runtime.OperationContext;
+import org.mule.module.extension.internal.runtime.OperationContextAdapter;
+
+/**
+ * Base class for {@link ArgumentResolver} implementations which depend
+ * on metadata from the current {@link MuleEvent} or {@link MuleMessage}
```

Comment:
that depend on each implementation. Ideally, I would like to be able to build DataSense support on top of the same mechanism introduced in this PR